### PR TITLE
chore (Algebra.Basic): make `RingHom.id` and related declarations reducible 

### PR DESCRIPTION
**Pull request by mattrobball**
These instances often appear as parameters to typeclass instances that need to unify reducibly.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

File: Mathlib/Algebra/Algebra/Basic.lean [hunk 1]
```diff
@@ -431,7 +432,7 @@ theorem coe_linearMap : ⇑(Algebra.linearMap R A) = algebraMap R A :=
   rfl
 #align algebra.coe_linear_map Algebra.coe_linearMap
 
-instance id : Algebra R R :=
+@[reducible] instance id : Algebra R R :=
```
Comment (eric-wieser) [inline Mathlib/Algebra/Algebra/Basic.lean:435]: I thought the whole point in `instance` is that it's reducible enough for anything typeclass related?

Comment (mattrobball) [reply Mathlib/Algebra/Algebra/Basic.lean:435]: Apparently not

Comment (eric-wieser) [reply Mathlib/Algebra/Algebra/Basic.lean:435]: Do you have a mwe to demo that?

(I agree with the principle behind the rest of this PR to change the non-instances to be reducible)

Comment (mattrobball) [reply Mathlib/Algebra/Algebra/Basic.lean:435]: Ah processing what you are saying now. You are correct, that is silly.

**Pull request closed without merging**